pkgs/development/compilers/temurin/default.nix

```diff
@@ -0,0 +1,444 @@
+  openjdk17,
+  openjdk21,
+  openjdk23,
+  openjdk24,
+  cups,
+  freetype,
+  alsa-lib,
+  libjpeg,
+  giflib,
+  libpng,
+  zlib,
+  lcms2,
+  fontconfig,
+  xorg,
+  cctools,
+  darwin,
+}:
+
+let
+  mkTemurinJDK =
+    {
+      featureVersion,
+      version,
+      build,
+      hash,
+    }:
+    let
+      bootstrapJdk =
+        if featureVersion == "11" then
+          openjdk11
+        else if lib.versionAtLeast featureVersion "24" then
+          openjdk24
+        else if lib.versionAtLeast featureVersion "23" then
+          openjdk23
+        else if lib.versionAtLeast featureVersion "21" then
+          openjdk21
+        else
+          openjdk17;
+
+      repoName = "jdk${featureVersion}u";
+      tagName = "jdk-${version}+${build}";
+
+      buildConfigMap = {
+        x86_64-linux =
+          if featureVersion == "11" then
+            "linux-x86_64-normal-server-release"
+          else
+            "linux-x86_64-server-release";
+        aarch64-linux =
+          if featureVersion == "11" then
+            "linux-aarch64-normal-server-release"
+          else
+            "linux-aarch64-server-release";
+        x86_64-darwin =
+          if featureVersion == "11" then
+            "macosx-x86_64-normal-server-release"
+          else
+            "macosx-x86_64-server-release";
+        aarch64-darwin =
+          if featureVersion == "11" then
+            "macosx-aarch64-normal-server-release"
+          else
+            "macosx-aarch64-server-release";
+      };
+
+      buildConfig = buildConfigMap.${stdenv.hostPlatform.system};
+
+    in
+    stdenv.mkDerivation (finalAttrs: {
+      pname = "temurin-jdk-${featureVersion}";
+      version = "${version}+${build}";
+
+      src = fetchFromGitHub {
+        owner = "adoptium";
+        repo = repoName;
+        rev = tagName;
+        inherit hash;
+      };
+
+      nativeBuildInputs =
+        [
+          autoconf
+          bash
+          which
+          zip
+          unzip
+          pkg-config
+          cpio
+          file
+          bootstrapJdk
+        ]
+        ++ lib.optionals stdenv.isDarwin [
+          cctools
+          darwin.bootstrap_cmds
+          darwin.xattr
+          darwin.DarwinTools
+          darwin.sigtool
+        ];
+
+      buildInputs =
+        [
+          cups
+          freetype
+          libjpeg
+          giflib
+          libpng
+          zlib
+          lcms2
+          fontconfig
+        ]
+        ++ lib.optionals stdenv.isLinux [
+          alsa-lib
+          xorg.libX11
+          xorg.libXext
+          xorg.libXrender
+          xorg.libXrandr
+          xorg.libXtst
+          xorg.libXt
+          xorg.libXi
+          xorg.xorgproto
+        ];
+
+      configurePhase = ''
+        runHook preConfigure
+
+        if [ "${featureVersion}" = "21" ] || [ "${featureVersion}" = "23" ] || [ "${featureVersion}" = "24" ]; then
+          export SOURCE_DATE_EPOCH=315532802
+        fi
+
+        bash configure \
```

Comment:
`bash configure` ?

Reply:
Hi,

Is because you need to run 'bash configure' to create a configuration.

Best Regards

Reply:
There's only one occurrence of `bash configure` in `nixpkgs`, that led me to think this is probably not the best practice? Perhaps I am wrong too.

When I look at https://github.com/adoptium/jdk21u/blob/master/configure, I see `#!/bin/bash` which obviously doesn't exist on NixOS and the need of manually doing `bash configure`.

Perhaps it would be a good idea to change the shebang in a `preConfigure` step ?